dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/QaServiceTest.kt

```diff
@@ -0,0 +1,47 @@
+package org.dataland.e2etests.tests
+
+import org.awaitility.Awaitility.await
+import org.dataland.datalandbackend.openApiClient.model.CompanyAssociatedDataEuTaxonomyDataForNonFinancials
+import org.dataland.datalandbackend.openApiClient.model.QAStatus as BackendQaStatus
+import org.dataland.datalandqaservice.openApiClient.model.QAStatus as QaServiceQaStatus
+import org.dataland.e2etests.auth.TechnicalUser
+import org.dataland.e2etests.utils.ApiAccessor
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import java.util.concurrent.TimeUnit
+
+class QaServiceTest {
+    private val apiAccessor = ApiAccessor()
+
+    private val testDataEuTaxonomyNonFinancials = apiAccessor.testDataProviderForEuTaxonomyDataForNonFinancials
+        .getTData(1).first()
+
+    private val testCompanyInformation = apiAccessor.testDataProviderForEuTaxonomyDataForNonFinancials
+        .getCompanyInformationWithoutIdentifiers(1).first()
+
+    @Test
+    fun `post dummy data, qa it and check the qa status changes`() {
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
+        val storedCompanyInfos = apiAccessor.companyDataControllerApi.postCompany(testCompanyInformation)
+        val companyAssociatedEuTaxonomyNonFinancialsData =
+            CompanyAssociatedDataEuTaxonomyDataForNonFinancials(
+                storedCompanyInfos.companyId,
+                "",
+                testDataEuTaxonomyNonFinancials,
+            )
+
+        val dataId = apiAccessor.dataControllerApiForEuTaxonomyNonFinancials
+            .postCompanyAssociatedEuTaxonomyDataForNonFinancials(
+                companyAssociatedEuTaxonomyNonFinancialsData, false,
+            ).dataId
+
+        assertEquals(BackendQaStatus.pending, apiAccessor.metaDataControllerApi.getDataMetaInfo(dataId).qaStatus)
+
+        val qaServiceController = apiAccessor.qaServiceControllerApi
+        await().atMost(10, TimeUnit.SECONDS)
+            .until { qaServiceController.getUnreviewedDatasets().contains(dataId) }
+        qaServiceController.assignQualityStatus(dataId, QaServiceQaStatus.rejected)
+
+        assertEquals(BackendQaStatus.rejected, apiAccessor.metaDataControllerApi.getDataMetaInfo(dataId).qaStatus)
+    }
```

Comment:
We should also test all the new "visibility" settings in Dataland.   Some of them that come to my mind:

- Assert that you get an exception if you are not a reviewer and try to modify the qaStatus of a dataId
- Assert that "pending" datasets can ONLY be seen by reviewer and the uploader
-... => anything else that comes to your mind?

Reply:
Florian und ich haben festgehalten, dass das im Team besporhcen werden sollte.

1) Sollten pending datasets dort überhaupt zu sehen sein?
2) Wenn ja, wie sollte das aussehen?

Reply:
I implemented a test for a reviewer trying to modify an already reviewed dataset.

Reply:
Would an admin or reviewer want to see all pending datasets though?